libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3404,7 +3404,7 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 				solAssert(toCategory == Type::Category::FixedBytes, "Invalid type conversion requested.");
 				body =
 					Whiskers("converted := <clean>(value)")
-					("clean", cleanupFunction(from))
+					("clean", cleanupFunction(_to))
```

Comment:
What this should really do is probably to use the cleanup function of the smaller of the types ``from`` and ``_to``.

Can you verify that we have tests for the following and they don't break:
```
ontract C {
    function g() external returns (bytes32 y) {
        bytes1 x;
        assembly { x := sub(0,1) }
        y = x;
    }

}
// ====
// compileViaYul: also
// ----
// g() -> 0xff00000000000000000000000000000000000000000000000000000000000000
```
?